sql/physicalplan: TestSpanResolverUsesCaches fails under race: expected replica (1,68) but got: (2,68)

sql/physicalplan.TestSpanResolverUsesCaches [failed](https://tanzanite.cluster.engflow.com/invocations/default/389e6840-3599-499a-b0e2-9cd00230a383?testReportRun=3&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cvc3FsL3BoeXNpY2FscGxhbjpwaHlzaWNhbHBsYW5fdGVzdA==) on master @ [e2e88aa288c81282b20ca0640da983f97b234324](https://github.com/cockroachdb/cockroach/commits/e2e88aa288c81282b20ca0640da983f97b234324):


```
=== RUN   TestSpanResolverUsesCaches
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestSpanResolverUsesCaches1302758279
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #108763 (expected label: C-bug)

pkg/sql/physicalplan_test/pkg/sql/physicalplan/span_resolver_test.go:124: (TestSpanResolverUsesCaches)
	NOTICE: .GetFirstStoreID() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().GetFirstStoreID() instead.

pkg/sql/physicalplan_test/pkg/sql/physicalplan/span_resolver_test.go:156: (TestSpanResolverUsesCaches)
	NOTICE: .NodeID() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().NodeID() instead.

pkg/sql/physicalplan_test/pkg/sql/physicalplan/span_resolver_test.go:156: (TestSpanResolverUsesCaches)
	NOTICE: .NodeID() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().NodeID() instead.
    span_resolver_test.go:159: expected replica (1,68) but got: (2,68)
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestSpanResolverUsesCaches1302758279
--- FAIL: TestSpanResolverUsesCaches (48.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSpanResolverUsesCaches.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34826

I am able to hit this on a gceworker, using:

```
./dev test pkg/sql/physicalplan --filter=TestSpanResolverUsesCaches --race --stress
```

Here's what the failure looks like:

```
FAIL: //pkg/sql/physicalplan:physicalplan_test (run 970 of 1000) (see /home/michae2/.cache/bazel/_bazel_michae2/88dadb79cb9eca13f5ad7e068969d8c1/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/testlogs/pkg/sql/physicalplan/physicalplan_test/run_970_of_1000/test.log)
ERROR: /home/michae2/go/src/github.com/cockroachdb/cockroach/pkg/sql/physicalplan/BUILD.bazel:41:8: Testing //pkg/sql/physicalplan:physicalplan_test (run 970 of 1000) failed: Test failed, aborting
==================== Test output for //pkg/sql/physicalplan:physicalplan_test (run 970 of 1000):
initialized metamorphic constant "max-batch-size" with value 2428
initialized metamorphic constant "max-batch-byte-size" with value 12736091
initialized metamorphic constant "parse-json-impl" with value 0
initialized metamorphic constant "coldata-batch-size" with value 1402
initialized metamorphic constant "default-batch-bytes-limit" with value 7532
initialized metamorphic constant "COCKROACH_BTREE_SPAN_FRONTIER_ENABLED" with value false
initialized metamorphic constant "kv.snapshot.ingest_as_write_threshold" with value apply_to_all
initialized metamorphic constant "spilling-queue-initial-len" with value 12
initialized metamorphic constant "merge-joiner-groups-buffer" with value 8
initialized metamorphic constant "row-container-rows-per-chunk-shift" with value 1
initialized metamorphic constant "lock-table-iters-before-seek" with value 2
initialized metamorphic constant "mvcc-incremental-iter-tbi" with value false
initialized metamorphic constant "kv.snapshot_receiver.excise.enabled" with value false
initialized metamorphic constant "storage.ingest_split.enabled" with value false
initialized metamorphic constant "mvcc-max-iters-before-seek" with value 0
initialized metamorphic constant "addsst-rewrite-forced" with value true
initialized metamorphic constant "raft-log-truncation-clearrange-threshold" with value 672502
initialized metamorphic constant "kv.snapshot.ingest_as_write_threshold" with value 0
initialized metamorphic constant "kv.expiration_leases_only.enabled" with value true
initialized metamorphic constant "datum-row-converter-batch-size" with value 1
initialized metamorphic constant "direct-scans-enabled" with value true
initialized metamorphic constant "ColIndexJoin-batch-size" with value 1144335
initialized metamorphic constant "ColIndexJoin-using-streamer-batch-size" with value 7208490
initialized metamorphic constant "parallel-scan-result-threshold" with value 6681
initialized metamorphic constant "inverted-joiner-batch-size" with value 1
initialized metamorphic constant "zig-zag-joiner-batch-size" with value 1
initialized metamorphic constant "resolve-membership-single-scan-enabled" with value false
initialized metamorphic constant "copy-batch-size" with value 9109
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 6
initialized metamorphic constant "tail-call-optimization-enabled" with value false
I231227 23:52:34.023484 1 (gostd) rand.go:250  [-] 1  random seed: -7728224196252952009

pkg/sql/physicalplan_test/pkg/sql/physicalplan/span_resolver_test.go:124: (TestSpanResolverUsesCaches)
        NOTICE: .GetFirstStoreID() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().GetFirstStoreID() instead.

pkg/sql/physicalplan_test/pkg/sql/physicalplan/span_resolver_test.go:156: (TestSpanResolverUsesCaches)
        NOTICE: .NodeID() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().NodeID() instead.

pkg/sql/physicalplan_test/pkg/sql/physicalplan/span_resolver_test.go:156: (TestSpanResolverUsesCaches)
        NOTICE: .NodeID() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().NodeID() instead.
test logs left over in: /tmp/cockroach/_tmp/e63e1750b419a6b7be00161aa6f9b319/logTestSpanResolverUsesCaches2567757200
--- FAIL: TestSpanResolverUsesCaches (60.61s)
    test_log_scope.go:170: test logs captured to: /tmp/cockroach/_tmp/e63e1750b419a6b7be00161aa6f9b319/logTestSpanResolverUsesCaches2567757200
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #108763 (expected label: C-bug)
    span_resolver_test.go:159: expected replica (1,68) but got: (2,68)
    panic.go:541: -- test log scope end --
FAIL
I231227 23:53:34.990795 1 (gostd) testmain.go:97  [-] 1  Test //pkg/sql/physicalplan:physicalplan_test exited with error code 1
```

--- a/pkg/sql/physicalplan/BUILD.bazel
+++ b/pkg/sql/physicalplan/BUILD.bazel
@@ -82,6 +82,7 @@ go_test(
         "//pkg/testutils/distsqlutils",
         "//pkg/testutils/physicalplanutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/leaktest",

--- a/pkg/sql/physicalplan/span_resolver_test.go
+++ b/pkg/sql/physicalplan/span_resolver_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/randgen"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -39,6 +40,8 @@ import (
 func TestSpanResolverUsesCaches(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "the test rarely flakes under race")
 
 	ctx := context.Background()
 	tc := testcluster.StartTestCluster(t, 4,

